include/igl/ASSERT.h

```diff
@@ -0,0 +1,9 @@
+// https://stackoverflow.com/a/985807/148668
+#include <cassert>
+#ifndef ASSERT
+#ifdef NDEBUG
+#define ASSERT(x) do { (void)sizeof(x);} while (0)
```

Comment:
I would suggest to prefix this with `IGL_`. You never know what other codebase might be defining their own `ASSERT` macro, in ways that could be incompatible with the libigl one...